src/components/Configure/content/fields/FieldMappings/RequiredFieldMappings.tsx

```diff
@@ -31,10 +34,23 @@ export function RequiredFieldMappings() {
   };
 
   const integrationFieldMappings = useMemo(
-    () => configureState?.read?.requiredMapFields?.filter(
-      isIntegrationFieldMapping,
-    ) || [],
-    [configureState],
+    () => {
+      if (!selectedObjectName || !fieldMapping) return [];
+      const dynamicIntegrationFieldMappings = fieldMapping ? Object.values(fieldMapping[selectedObjectName] || {}).flat() : [];
```

Comment:
nit: eslint code style says a bit too long; break into a few lines for readability.